### PR TITLE
GCW-3403- Adding Missing actions on history tab 

### PR DESCRIPTION
**Pull request by abulsayyad123**
### Ticket Link: 

https://jira.crossroads.org.hk/browse/GCW-3403

### What does this PR do?

This PR adds Designated action in history.

### CODE REVIEW

File: app/controllers/items/history.js [hunk 1]
```diff
@@ -59,11 +63,17 @@ export default detail.extend({
 
         const newGroup = {
           key: groupKey,
-          type: isItemAction ? action.get("action").capitalize() : "Edited",
+          type: isItemAction
+            ? action.get("action").capitalize()
```
Comment (bharat619) [inline app/controllers/items/history.js:67]: Its better to simplify this. Too much nesting for a single statement

File: app/models/orders_package.js [hunk 2]
```diff
@@ -33,6 +38,14 @@ export default Model.extend({
     return this.get("designation.code");
   }),
 
+  locationSubscript: Ember.computed("quantity", function() {
+    return Number(this.get("quantity")) >= 0 ? "To" : "From";
+  }),
+
+  arrow: Ember.computed("quantity", function() {
```
Comment (bharat619) [inline app/models/orders_package.js:45]: is this new computed property used somewhere?
Im unable to find it 🤔

File: app/controllers/items/history.js [hunk 3]
```diff
@@ -59,11 +63,17 @@ export default detail.extend({
...
+            : action.get("state")
+            ? action.get("state")
+            : "Edited",
```
Comment (swatijadhav) [inline app/controllers/items/history.js:70]:
```suggestion
          type: isItemAction
            ? action.get("action").capitalize()
            : (action.get("state") || "Edited"),
```

File: app/controllers/items/history.js [hunk 4]
```diff
@@ -59,11 +63,17 @@ export default detail.extend({
...
           date: createdAt,
           user: isItemAction
             ? action.get("user.fullName")
-            : action.get("whodunnitName"),
+            : action.get("whodunnitName")
+            ? action.get("whodunnitName")
+            : action.get("updatedBy.fullName"),
```
Comment (swatijadhav) [inline app/controllers/items/history.js:76]:
```suggestion
            : (action.get("whodunnitName") || action.get("updatedBy.fullName")),
```

File: app/controllers/items/history.js [hunk 5]
```diff
@@ -36,10 +36,14 @@ export default detail.extend({
...
       )}/${createdAt}`;
-    } else {
+    } else if (action.get("whodunnitName")) {
       return `${action.get("whodunnitName")}/${this.get(
         "model.state"
       )}/${createdAt}`;
+    } else {
+      return `${this.get("session.currentUser.fullName")}/${action.get(
+        "state"
+      )}/${createdAt}`;
```
Comment (swatijadhav) [inline app/controllers/items/history.js:46]:
```suggestion
    } 
    
    if (action.get("whodunnitName")) {
      return `${action.get("whodunnitName")}/${this.get(
        "model.state"
      )}/${createdAt}`;
    } 
    
    return `${this.get("session.currentUser.fullName")}/${action.get(
        "state"
      )}/${createdAt}`;
```

File: app/models/orders_package.js (hunk 2 above)
Comment (abulsayyad123) [reply app/models/orders_package.js:45]: It is used in hbs, it is added for this orders_packages.

**Review by swatijadhav — APPROVED** [review]

**Review by bharat619 — APPROVED** [review]

**Pull request merged**